### PR TITLE
Copy rimage toml config files to SOF repository

### PR DESCRIPTION
**Pull request by aiChaoSONG**
This patch moves rimage/config to sof/config.
Firmware manifest configuration files should stay
with SOF instead of rimage tool.

Link: https://github.com/thesofproject/sof/issues/7270

If you don't think sof/config is a good place, please comment the place you think is good.

### CODE REVIEW

**Comment by marc-hb** [comment]
I'd like to finish https://github.com/zephyrproject-rtos/zephyr/pull/54700 first.

There's also the question about whether it's good idea to keep growing the role of a user-space tool written in a low-level, unsafe and unreliable language.

**Comment by aiChaoSONG** [comment]
I think this patch will impact the commit https://github.com/zephyrproject-rtos/zephyr/pull/54700/commits/810b0f273b8586bba99446228bc9e8364b468a38, but I think this one should go first, because in your commit, you use sof/rimage/config as the default dir, which is going to change very soon. this patch continues using `west sign` which I don't think you changed any option for it.

**Comment by kv2019i** [comment]
@aiChaoSONG Thanks for doing this! Not sure why System/merge/build failed, but otherwise CI looks very promising.

@marc-hb Ack on #54700  but given it didn't make it to SOF2.5 and this PR is going to be very labour-intensive to keep up-to-date (as we have a continuous stream of rimage PRs to these toml files), I'd like to timebox it so that if #54700 is not done this week, we'll switch the priorities and do the rimage move first. This will have immediate benefits as the wave of rimage PRs is jus increasing.

**Comment by aiChaoSONG** [comment]
SOFCI TEST

**Comment by marc-hb** [comment]
```diff
- [RFC] Move rimage toml config files to SOF repository
+ [RFC] Copy rimage toml config files to SOF repository
```
I guess the removal will come later to ease the transition?

What is the transition strategy?

**Comment by aiChaoSONG** [comment]
Update: 
- commit message revision, move -> copy because we don't delete rimage/config yet
- sync with rimage main, add newly merged rimage changes.

**Comment by aiChaoSONG** [comment]
> What is the transition strategy? Don't forget there are two use cases:

- I think zephyr/west.yml doesn't use rimage.
- sof/west.yml, once the PR merged, we are using tomls in sof/config. we can keep the rimage commit currently used as long as there are no changes to rimage itself, or we can update to later rimage main.

**Comment by marc-hb** [comment]
> I think zephyr/west.yml doesn't use rimage.

Internal zephyr CI does use rimage, it's required to test on hardware.

**Comment by gkbldcig** [comment]
Can one of the admins verify this patch?

**Comment by kv2019i** [comment]
@aiChaoSONG This can proceed now, but we need to keep the toml files in rimage as well to allow for a smooth transition.

File: config/bdw.toml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+version = [1, 0]  # use simple file write
```
Comment (lgirdwood) [inline config/bdw.toml:1]: Intel legacy platforms are no longer in main branch - i.e. cavs2.5 and above is for main.

**Comment by aiChaoSONG** [comment]
Close as we don't want a simple toml file moving, instead, we want to split platform config and algorithm config

**Pull request closed without merging**